src/main/java/frc/robot/subsystems/swerve/SwerveModule.java

```diff
@@ -29,20 +31,24 @@
     public SwerveModule(SwerveModuleConstants moduleConstants) {
         this.constants = moduleConstants;
         this.driveFeedforward = new TrigonPIDController(
-                moduleConstants.drivePIDCoefs);
+                moduleConstants.drivePIDFCoefs);
         encoderOffset = moduleConstants.encoderOffset;
 
         /* Angle Encoder Config */
-        angleEncoder = new TrigonTalonSRX(moduleConstants.encoderID);
-        configAngleEncoder();
+        angleMotor = new TrigonPIDFTalonFX(moduleConstants.angleMotorID, SwerveConstants.ANGLE_MOTOR_CONFIG,
+                moduleConstants.anglePIDFCoefs);
+        configAngleMotor();
 
         /* Angle Motor Config */
-        mAngleMotor = new TrigonTalonFX(moduleConstants.angleMotorID);
-        configAngleMotor();
+        driveMotor = new TrigonPIDFTalonFX(moduleConstants.driveMotorID, SwerveConstants.DRIVE_MOTOR_CONFIG,
+                moduleConstants.anglePIDFCoefs);
+        configDriveMotor();
 
         /* Drive Motor Config */
-        mDriveMotor = new TrigonTalonFX(moduleConstants.driveMotorID);
-        configDriveMotor();
+        angleEncoder = new TrigonTalonSRX(
+                moduleConstants.angleMotorID,
+                new MotorConfig(new MotorConfig(), false, SwerveConstants.ENCODER_INVERT));
```

Comment:
why use a copy constructor when you don't have an existing object?

Reply:
shortest constructor to use. as we don't need the other config setting i give a blank config and the inverts. i could add *another* constrcutor but that's a bit too much